packages/kit-tamagui-toast/src/index.ts

```diff
@@ -1,20 +1,24 @@
-import type { ActionDefinition } from '@rise-tools/react'
+import type { ActionsDefinition } from '@rise-tools/react'
 import { useToastController } from '@tamagui/toast'
+import z from 'zod'
 
-import type { toast } from './server'
+import type { ToastAction } from './server'
 
-type ToastActions = {
-  toast: ActionDefinition<ReturnType<typeof toast>>
-}
+const ToastActionPayload = z.object({
+  title: z.string(),
+  message: z.string().optional(),
+  duration: z.number().optional(),
+})
 
-export const useToastActions = (): ToastActions => {
+export const useToastActions = (): ActionsDefinition<ToastAction> => {
   const toast = useToastController()
 
   return {
     toast: {
       action: ({ title, ...options }) => {
         toast.show(title, options)
       },
+      validate: (payload) => ToastActionPayload.parse(payload),
```

Comment:
With `typia`, we will be able to write:
```
typia.assert<ActionPayload<ToastAction>>(input)
```
and call it a day!

Reply:
cool, if the build step works! Also its a bit opaque to see the runtime code

Reply:
Yeah. For now, I am not continuing with it. I don't like patching TypeScript and it will not scale for users that ever run from source (and haven't patched their TypeScript). For things such as simple actions, we can probably write some basic if conditions ourselves. 

Anyways, it's a nice problem to have and I think we're aware of it, but it's definitely too early to look into it. Zod has nice DX and isn't going to be too hard to upgrade this way or another.